models/dsn_helpers.go

```diff
@@ -40,30 +40,55 @@ func FindDSNByServiceIDandPMMAgentID(q *reform.Querier, serviceID, pmmAgentID, d
 		return "", err
 	}
 
-	var agentType AgentType
-	switch svc.ServiceType {
-	case MySQLServiceType:
-		agentType = MySQLdExporterType
-	case MongoDBServiceType:
-		agentType = MongoDBExporterType
-	case PostgreSQLServiceType:
-		agentType = PostgresExporterType
-	default:
-		return "", status.Errorf(codes.FailedPrecondition, "Couldn't resolve dsn, as service is unsupported")
-	}
-
-	fexp, err := FindAgents(q, AgentFilters{
+	agents, err := FindAgents(q, AgentFilters{
 		ServiceID:  serviceID,
-		AgentType:  &agentType,
 		PMMAgentID: pmmAgentID,
 	})
 	if err != nil {
 		return "", err
 	}
 
-	if len(fexp) != 1 {
-		return "", status.Errorf(codes.FailedPrecondition, "Couldn't resolve dsn, as there should be only one exporter")
+	types := make(map[AgentType]*Agent)
+	for _, a := range agents {
+		if _, ok := types[a.AgentType]; ok {
```

Comment:
We will get error in case we have two mongodb_exporters and one qan-mongodb-profiler for Service1 and we run MongoDBExplain action.